roles/grub/tasks/main.yml

```diff
@@ -5,6 +5,13 @@
   when: cgroupsv2_disabled is defined and cgroupsv2_disabled is true
   tags: ['always']
 
+- name: Update GRUB configuration for hugepages
+  ansible.builtin.copy:
+    content: "GRUB_CMDLINE_LINUX_DEFAULT=\"{{ grub_config.cmdline_linux_default | join(' ') }}\""
```

Comment:
`grub_config.cmdline_linux_default` is variable which may not be defined (above you define it as the group_vars vairable for subset of nodes), therefore I'd except `when` keyword in this task similar to the task above (line 5)